### PR TITLE
Add network status. Closes #168

### LINKED ISSUE #168: Connected/disconnected status indicator
Figma: [Connected/disconnected status indicator - #168](https://www.figma.com/file/3iMRh8xBPrT0UU8d8W5Wh0/Connected-disconnected-status-indicator?node-id=0%3A1)

Specs:
TBD

### PR DESCRIPTION
**Pull request by juliangruber**
image

image

The network status refreshes once a second and can show

- `red`: `Disconnected`
- `yellow`: `Connected, but not holepunchable`
- `green`: `Connected`

Closes #168

### CODE REVIEW

**Comment by jameslibscie** [comment]
If the user's connection is yellow / not holepunchable, what can they do to remedy this?

**Comment by juliangruber** [comment]
> If the user's connection is yellow / not holepunchable, what can they do to remedy this?

It's usually firewalls and/or network hardware and setup.

**Comment by jameslibscie** [comment]
> > If the user's connection is yellow / not holepunchable, what can they do to remedy this?
> 
> It's usually firewalls and/or network hardware and setup.

Thanks - and what's the consequence?

**Comment by chartgerink** [comment]
Obligatory GIF

File: app/components/menu/menu.js [hunk 1]
```diff
@@ -90,14 +107,54 @@ const ButtonNavLink = ({ history, to, ...props }) => (
...
+  }, [networkStatus])
 
   return (
     <Container>
       <StyledLogo onClick={() => history.push('/')} />
+      <NetworkStatusContainer
+        title={
+          {
+            green: 'Connected',
+            yellow: 'Connected, but not holepunchable',
```
Comment (jameslibscie) [inline app/components/menu/menu.js:145]: Want something a bit more user-friendly for this. Open to other suggestions :)

```suggestion
            yellow: 'Connection limited (not holepunchable) - please check your firewall and network setup',
```

**Comment by chartgerink** [comment]
> Works well! Tried disconnecting and stuff and it worked perfectly.
> 
> However, I couldn't see any tooltips upon hover.

I thought this at first as well, but the hover takes a second to appear and really is there :D

**Comment by juliangruber** [comment]
@jameslibscie more research needs to be done on holepunching. Are you ok with opening a separate issue for it and merging this as is?

**Comment by jameslibscie** [comment]
@juliangruber I don't like the word "holepunchable" being the primary indicator of what the problem is, since it's a word that no one knows and doesn't yield usable Google results. I.e., it doesn't at all help the user to have that word there right now. I can see how it's hard to provide an accurate description for this that is helpful to the user, since we don't really know the consequences of the status being yellow.

How about:
> Connection may be limited, please check your firewall and network setup if you are experiencing issues

**Comment by juliangruber** [comment]
> since it's a word that no one knows and doesn't yield usable Google results.

Ah, maybe we should change it to "Hole punching unavailable" https://en.wikipedia.org/wiki/Hole_punching_(networking)

**Comment by juliangruber** [comment]
>  it's a word that no one knows

This is an assumption. If you've used bittorrent clients before for example chances are high you've run across this, figuring out networking issues.

**Comment by jameslibscie** [comment]
> > since it's a word that no one knows and doesn't yield usable Google results.
> 
> Ah, maybe we should change it to "Hole punching unavailable" https://en.wikipedia.org/wiki/Hole_punching_(networking)

Ah great, there are resources!

How about this then?
> Connection may be limited - hole punching unavailable

**Comment by juliangruber** [comment]
> Connection may be limited - hole punching unavailable

Without hole punching the connection is definitely limited so what about

> Connection limited - hole punching unavailable

**Comment by jameslibscie** [comment]
> > Connection may be limited - hole punching unavailable
> 
> Without hole punching the connection is definitely limited so what about
> 
> > Connection limited - hole punching unavailable

Perfect!

**Pull request merged**